fleetctl/fleetctl.go

```diff
@@ -663,6 +671,85 @@ func createUnit(name string, uf *unit.UnitFile) (*schema.Unit, error) {
 	return &u, nil
 }
 
+// checkReplaceUnitState checks if the unit should be replaced.
+// It takes a Unit object as a parameter.
+// It returns 0 on success and if the unit should be replaced, 1 if the
+// unit should not be replaced; and any error encountered.
+func checkReplaceUnitState(unit *schema.Unit) (int, error) {
+	// We replace units only for 'submit', 'load' and
+	// 'start' commands.
+	allowedReplace := map[string][]job.JobState{
+		"submit": []job.JobState{
+			job.JobStateInactive,
+		},
+		"load": []job.JobState{
+			job.JobStateInactive,
+			job.JobStateLoaded,
+		},
+		"start": []job.JobState{
+			job.JobStateInactive,
+			job.JobStateLoaded,
+			job.JobStateLaunched,
+		},
+	}
+
+	if allowedJobs, ok := allowedReplace[currentCommand]; ok {
+		for _, j := range allowedJobs {
+			if job.JobState(unit.DesiredState) == j {
+				return 0, nil
+			}
+		}
+		stderr("Warning: can not replace Unit(%s) in state '%s', use the appropriate command", unit.Name, unit.DesiredState)
+	} else {
+		// This function should only be called from 'submit',
+		// 'load' and 'start' upper paths.
+		return 1, fmt.Errorf("error: replacing units is not supported in this context")
+	}
+
+	return 1, nil
+}
+
+// checkUnitCreation checks if the unit should be created.
+// It takes a unit file path as a parameter.
+// It returns 0 on success and if the unit should be created, 1 if the
+// unit should not be created; and any error encountered.
+func checkUnitCreation(arg string) (int, error) {
+	name := unitNameMangle(arg)
+
+	// First, check if there already exists a Unit by the given name in the Registry
+	unit, err := cAPI.Unit(name)
+	if err != nil {
+		return 1, fmt.Errorf("error retrieving Unit(%s) from Registry: %v", name, err)
+	}
+
+	// check if the unit is running
+	if unit == nil {
+		if sharedFlags.Replace {
+			log.Debugf("Unit(%s) was not found in Registry", name)
+		}
+		// Create a new unit
+		return 0, nil
+	}
+
+	// if sharedFlags.Replace is not set then we warn
+	different, err := isLocalUnitDifferent(arg, unit, !sharedFlags.Replace, false)
```

Comment:
> Part of the confusion is also because you now do the warning handling inside isLocalUnitDifferent(), but the "fatal" error handling outside. If you could somehow shuffle them around to be in the same place (I don't really care which, as long as it's the same), that would certainly improve readability.

I can't see here where "warn if units differ" relates to fatal error ?!
And we check if local unit is either a simple unit or an instance of template unit